### PR TITLE
Fix cross compilation errors

### PR DESCRIPTION
**Pull request by calvinlfer**
Hey @Grryum thank you for your excellent work on integrating micrometer with zio-metrics-connectors
I managed to take a look and did some slight re-organization of the code inside MicrometerMetricListener and also introduced MicroMeterConfig so we didn't have to hardcode the summary percentile digits. I also made AtomicDouble a value class to see if we can further drop overhead. I have also gone ahead and fixed cross compilation issues that were causing the original PR to fail. Hope you don't mind 😺

### CODE REVIEW

File: micrometer/src/main/scala/zio/metrics/connectors/micrometer/MicroMeterConfig.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package zio.metrics.connectors.micrometer
+
+final case class MicroMeterConfig(summaryPercentileDigitsOfPrecision: Int)
```
Comment (Grryum) [inline micrometer/src/main/scala/zio/metrics/connectors/micrometer/MicroMeterConfig.scala]: Very minor but let it be MicrometerConfig :)

File: micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala [hunk 2]
```diff
@@ -9,7 +9,7 @@ import io.micrometer.core.instrument.{MeterRegistry, Tag}
 
 package object micrometer {
 
-  lazy val micrometerLayer: ZLayer[MeterRegistry, Nothing, Unit] =
+  val micrometerLayer: ZLayer[MeterRegistry with MicroMeterConfig, Nothing, Unit] =
```
Comment (Grryum) [inline micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala]: why do you prefer to remove laziness here? I think it is ok to have lazy val layer

**Comment by Grryum** [comment]
@calvinlfer thank you for your improvements! I switched my context and didn't manage to finish MR properly, thanks for fixing cross compilation issues :)
I leave very minor comments, please take a look.

File: micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala (hunk 2 above)
Comment (calvinlfer) [reply micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala]: Layers are already lazy by default

Comment (Grryum) [reply micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala]: yes cause it is effect :) but constructing effect also need some allocation. Approach with lazy layer mensioned in ZIO documentation (https://zio.dev/reference/di/manual-layer-construction/). And similarly is used in other modules:
- https://github.com/zio/zio-metrics-connectors/blob/zio/series2.x/core/src/main/scala/zio/metrics/connectors/datadog/package.scala#L11
- https://github.com/zio/zio-metrics-connectors/blob/zio/series2.x/core/src/main/scala/zio/metrics/connectors/newrelic/package.scala#L9
- https://github.com/zio/zio-metrics-connectors/blob/zio/series2.x/core/src/main/scala/zio/metrics/connectors/prometheus/package.scala#L8
- https://github.com/zio/zio-metrics-connectors/blob/zio/series2.x/core/src/main/scala/zio/metrics/connectors/statsd/package.scala#L8

Thats why I decided to make layer lazy. On the other hand this is the single layer in package in separated module. So if user uses it as a dependency that definitely he will use this layer.

Comment (calvinlfer) [reply micrometer/src/main/scala/zio/metrics/connectors/micrometer/package.scala]: I've added it back

File: micrometer/src/main/scala/zio/metrics/connectors/micrometer/MicroMeterConfig.scala (hunk 1 above)
Comment (calvinlfer) [reply micrometer/src/main/scala/zio/metrics/connectors/micrometer/MicroMeterConfig.scala]: Done

**Pull request merged**